hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -182,24 +128,20 @@ checkAbort context@ScriptContext{scriptContextTxInfo = txInfo} headContext parti
 checkCollectCom ::
   -- | Script execution context
   ScriptContext ->
-  -- | Static information about the head (i.e. address, value, currency...)
-  HeadContext ->
   -- | Initial state
-  (ContestationPeriod, [Party]) ->
+  State ->
```

Comment:
This is a step back. It's less type safe. If you want to get rid of large tuples, we should introduce also on-chain the `InitialState` sub-type, i.e. ` data State = ... | Initial InitialState | ... `